wasm/execution/configuration.py

```diff
@@ -125,6 +129,23 @@ def __init__(self, store: Store) -> None:
         super().__init__(store)
         self.frame_stack = FrameStack()
 
+    def execute(self) -> Tuple[TValue, ...]:
+        # TODO: unwrap import once logic functions move out of wasm.main
+        from wasm.main import opcode2exec
+
+        while self.has_active_frame:
+            instruction = next(self.instructions)
+
+            logic_fn = opcode2exec[instruction.opcode][0]
+            logic_fn(self)
+
+        if len(self.result_stack) > 1:
+            raise Exception("Invariant?")
```

Comment:
Heh, maybe a bit more info? Could be length of the stack, items on it, etc.

Reply:
Updated to make it clear that the WASM spec only allows at most a single return value.  This is done as an `Invariant` using raw `Exception` because validation **should** always catch this before we could even reach this, so hitting this would indicate that there is something wrong with our implementation.